lib/wallet/http.js

```diff
@@ -478,6 +478,7 @@ class HTTP extends Server {
         subtractIndex: valid.i32('subtractIndex'),
         depth: valid.u32(['confirmations', 'depth']),
         template: valid.bool('template', sign),
+        changeAddress: valid.str(['changeAddress', 'change']),
```

Comment:
I think we'd need to validate the change address before we get here, the same way the receive address is validated a few lines later.

```diff
diff --git a/lib/wallet/http.js b/lib/wallet/http.js
index ae117499..43d1f173 100644
--- a/lib/wallet/http.js
+++ b/lib/wallet/http.js
@@ -466,6 +466,10 @@ class HTTP extends Server {
       const outputs = valid.array('outputs', []);
       const sign = valid.bool('sign', true);

+      let changeAddress = valid.str(['changeAddress', 'change']);
+      if (changeAddress)
+        changeAddress = Address.fromString(changeAddress, this.network);
+
       const options = {
         rate: valid.u64('rate'),
         blocks: valid.u32('blocks'),
@@ -478,7 +482,7 @@ class HTTP extends Server {
         subtractIndex: valid.i32('subtractIndex'),
         depth: valid.u32(['confirmations', 'depth']),
         template: valid.bool('template', sign),
-        changeAddress: valid.str(['changeAddress', 'change']),
+        changeAddress,
         outputs: []
       };
```

Reply:
Makes sense, done.